apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/TimerInterceptor.java

```diff
@@ -0,0 +1,68 @@
+package gov.cms.bfd.server.war;
+
+import ca.uhn.fhir.interceptor.api.Hook;
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.interceptor.api.Pointcut;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+
+/**
+ * With the use of HAPI's pointcut and interceptors, time metrics at various instances i.e.
+ * Pre-handling and outgoing response in the BFD API call lifecycle can be generated and logged
+ * {@link BfdMDC}. For more info on server pointcuts:
+ * https://hapifhir.io/hapi-fhir/docs/interceptors/server_pointcuts.html
+ */
+@Interceptor
+public class TimerInterceptor {
+
+  /** Pointcut to log time in milliseconds when a request is pre-processed. */
+  @Hook(Pointcut.SERVER_INCOMING_REQUEST_PRE_PROCESSED)
+  public void requestPreProcessed() {
+    // log current instance to MDC
+    Long currentTime = System.currentTimeMillis();
+    BfdMDC.put(
+        "hapi.server_incoming_request_pre_process_time_in_millis", Long.toString(currentTime));
```

Comment:
sorry didn't see this before -- I think these should be timestamp_in_millis (time could imply duration to some). Thanks for making the other changes!

Reply:
Updated